tests/system/common.py

```diff
@@ -819,3 +824,53 @@ def kill_process_on_host(hostname, pattern):
     else:
         print("Killed no pids")
     return pids
+
+
+@lru_cache()
+def dcos_masters_public_ips():
+    @retrying.retry(
+        wait_fixed=1000,
+        stop_max_attempt_number=240,  # waiting 20 minutes for exhibitor start-up
+        retry_on_exception=ignore_provided_exception(DCOSException))
+    def all_master_ips():
+        return get_all_master_ips()
+
+    master_private_ips = all_master_ips()
+
+    master_public_ips = []
+
+    for private_ip in master_private_ips:
+        result, ip = shakedown.run_command(private_ip, '/opt/mesosphere/bin/detect_ip_public')
+        master_public_ips.append(ip)
+
+    return master_public_ips
+
+
+# since exhibitor might take ~ 20 minutes to start up, we need to wait for it
+def wait_for_service_endpoint(service_name, timeout_sec=120):
+    """Checks the service url if available it returns true, on expiration
+    it returns false"""
+
+    @retrying.retry(
+            wait_fixed=1000,
+            stop_max_attempt_number=timeout_sec/5,  # underlying http.get has 5 seconds timeout, so we have to scale it
+            retry_on_exception=ignore_provided_exception(DCOSException))
+    def check_service_availability_on_master(master_ip, service):
+        url = "https://{}/service/{}/".format(master_ip, service)
+
+        toml_config = config.get_config()
+        auth_token = config.get_config_val("core.dcos_acs_token", toml_config)
+        auth = DCOSAcsAuth(auth_token)
+        response = requests.request(
+            method='get',
+            url=url,
+            timeout=5,
+            auth=auth,
+            verify=str(get_ca_file()))
```

Comment:
I think requests offers

```python
response = requests.get(url, timeout=5, auth=auth, verify=...)
if response.status_code != 200:
  raise DCOSException("service " + service_name + " is unavailable at " + master_ip)
```

Reply:
Also, please check for the CA file. Or does requests make the check? Can `verify` be `None`?